### ISSUE TITLE
[High-Level-Design] Spark Connect

### ISSUE THREAD

**Issue by FlowerBirds** [japila-books/spark-sql-internals#13]
https://docs.google.com/document/d/17X6-P5H2522SnE-gF1BVwyildp_PDX8oXD-4l9vqQmA/edit#heading=h.oml6js5zq00v

https://docs.google.com/document/d/1Mnl6jmGszixLW4KcJU5j9IgpG9-UabS0dcM6PM2XGDc/edit#heading=h.wmsrrfealhrj

**Comment by jaceklaskowski** [comment]
Hey @FlowerBirds 👋  Thanks for reporting the issue. Can you elaborate more what you'd like this issue to fix? 🙏

**Comment by FlowerBirds** [comment]
I see that this repository is about the principle of parsing Spark SQL. In the Spark SQL->Features->Spark Connect section, most of the content is still under development. I have recently been researching the usage of Spark Connect and have found some relevant design documents. I hope that you can incorporate them into the future improvements.

**Comment by jaceklaskowski** [comment]
It took me long(er) to be on it, but it is happening and got curious how exactly Spark Connect works under the covers. Starting with Spark Connect Client.

**Comment by jaceklaskowski** [comment]
I think https://books.japila.pl/spark-connect-internals/ is what you asked me for long ago, didn't you, @FlowerBirds?

**Comment by FlowerBirds** [comment]
Yes, that's what I said. Thank you for your contribution. It was used in a project for a while, and when version 4.0.0 is released, this feature will be used more widely.
